pipeline_tools/shared/submission/create_file_descriptor.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+import argparse
+import datetime
+import json
+import re
+
+from pipeline_tools.shared.submission.format_map import get_uuid5, convert_datetime
+
+
+def build_file_descriptor(
+    file_path,
+    size,
+    sha256,
+    crc32c,
+    creation_time,
+    raw_schema_url,
+    file_descriptor_schema_version,
+):
+    """Create the submission envelope in Ingest service.
+
+    Args:
+        file_path (str): Path to the described file.
+        size (str): Size of the described file in bytes.
+        sha256 (str): sha256 hash value of the described file.
+        crc32c (str): crc32c hash value of the described file.
+        creation_time (str): Timestamp of the creation time of the described file.
+        raw_schema_url (str): URL prefix for retrieving HCA metadata schemas.
+        file_descriptor_schema_version (str): Version of the metadata schema that the file_descriptor.json conforms to.
+    """
+
+    SCHEMA_TYPE = 'file_descriptor'
+    file_id = get_uuid5(get_uuid5(sha256))
+    file_version = convert_datetime(creation_time)
+
+    file_descriptor = {
+        'describedBy': get_file_descriptor_described_by(
+            schema_url=raw_schema_url, schema_version=file_descriptor_schema_version
+        ),
+        'schema_version': file_descriptor_schema_version,
+        'schema_type': SCHEMA_TYPE,
+        'content_type': 'application/unknown',  # TODO: Make this actually work
+        'size': int(size),
+        'sha256': sha256,
+        'crc32c': crc32c,
+        'file_id': file_id,
+        'file_version': file_version,
+        'file_name': get_relative_file_location(file_path),
+    }
+
+    return file_descriptor
+
+
+def get_datetime_from_file_info(file_info):
```

Comment:
Is this function being used anywhere?

Reply:
Good point! I think this got dropped during one of the iterations - removing now.